golos.publication/golos.publication.cpp

```diff
@@ -152,137 +148,96 @@ void publication::upvote(account_name voter, account_name author, std::string pe
     require_auth(voter);
     limit_of_votes(voter);
 
-    tables::vote_table vote_table(_self, author);
-    tables::voters_table voters_table(_self, author);
+//    eosio_assert(weight > 0, "The weight sign can't be negative.");
 
-    std::vector<structures::rshares> rshares;
+    set_vote(voter, author, permlink, weight);
+}
 
-    structures::rshares rshares_obj{0, 0, 0, 0};
+void publication::downvote(account_name voter, account_name author, std::string permlink, asset weight) {
+    require_auth(voter);
 
-    rshares.push_back(rshares_obj);
+//    eosio_assert(weight < 0, "The weight sign can't be positive.");
 
-    structures::post posttable_obj;
+    set_vote(voter, author, permlink, weight);
+}
 
-    if (get_post(author, permlink, posttable_obj)) {
-        auto votetable_index = vote_table.get_index<N(postid)>();
-        auto votetable_obj = votetable_index.find(posttable_obj.id);
-        if (votetable_obj != votetable_index.end()) {
-            auto voterstable_index = voters_table.get_index<N(postid)>();
-            auto voterstable_obj = voterstable_index.find(posttable_obj.id);
-            while (voterstable_obj != voterstable_index.end() &&
-                   posttable_obj.id == voterstable_obj->post_id) {
-                eosio_assert(voter != voterstable_obj->voter,
-                             "You have already voted for this post.");
-                ++voterstable_obj;
-            }
+void publication::unvote(account_name voter, account_name author, std::string permlink, asset weight) {
```

Comment:
weight is not needed to pass in this action, it's already known that `unvote` weight equal to 0